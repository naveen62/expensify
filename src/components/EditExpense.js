import React from  'react'
import {connect} from 'react-redux';
import ExpenseForm from './ExpenseForm';
import {startEditExpense, startRemoveExpense} from '../actions/expense'

export class EditExpense extends React.Component {
    onSubmit = (expense) => {
        this.props.startEditExpense(this.props.match.params.id, expense).then(() => {
            this.props.history.push('/')
        })
    }
    handleRemove = () => {
        this.props.startRemoveExpense(this.props.match.params.id).then(() => {
            this.props.history.push('/')
        })
    }

    render() {
        return (
            <div>
                <ExpenseForm
                 expense={this.props.expense}
                 onSubmit={this.onSubmit} />
                    <button onClick={this.handleRemove} >Remove</button>       
            </div>
        ) 
    }
}

const mapStateToProps = (state, props) => {
    return {
        expense: state.expenses.find((expense) => {
            return expense.id === props.match.params.id
        })
    }
}
const mapDispatchToProps = (dispatch) => {
    return {
        startEditExpense: (id, expense) => dispatch(startEditExpense(id, expense)),
        startRemoveExpense: (id) => dispatch(startRemoveExpense(id))
    }
}

export default connect(mapStateToProps, mapDispatchToProps)(EditExpense);
