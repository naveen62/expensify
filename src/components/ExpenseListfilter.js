import React from 'react';
import {connect} from 'react-redux';
import {setStartDate, setEndDate ,setTextFilter, sortByAmount, sortByDate } from "../actions/filter";
import {DateRangePicker} from 'react-dates'

export class ExpenseListfilter extends React.Component {
   state = {
       calendarFocused: null
   }
   onDatesChange = ({startDate, endDate}  ) => {
        this.props.setStartDate(startDate);
        this.props.setEndDate(endDate);
   }
   onFocusChange = (cal) => {
       this.setState(() => ({calendarFocused: cal}))
   }    
   onTextChange = (e) => {
    this.props.setTextFilter(e.target.value)
   }
   onSortChange = (e) => {
    if(e.target.value === 'date') {
        this.props.sortByDate()
    } else if(e.target.value === 'amount') {
        this.props.sortByAmount()
    }
    }
    render() {
        return (
            <div>
                <input type='text' value={this.props.filter.text} onChange={this.onTextChange} />
                <select onChange={this.onSortChange}>
                    <option value='date'>Date</option>
                    <option value='amount'>Amount</option>
                </select>
                <DateRangePicker
                    startDate={this.props.filter.startDate}
                    endDate={this.props.filter.endDate}
                    onDatesChange={this.onDatesChange}
                    focusedInput={this.state.calendarFocused}
                    onFocusChange={this.onFocusChange}
                    numberOfMonths={1}
                    showClearDates={true}
                    isOutsideRange={() => false}
                 /> 
            </div>
        )
    }
}



const mapsStateToProps = (state) => {
    return {
        filter: state.filter
    }
}
const mapDispatchToProps = () => ({
    setTextFilter: (text) => dispatch(setTextFilter(text)),
    sortByDate: () => dispatch(sortByDate()),
    sortByAmount: () => dispatch(sortByAmount()),
    setStartDate: (startDate) => dispatch(setStartDate(startDate)),
    setEndDate: (endDate) => dispatch(setEndDate(endDate))
})
export default connect(mapsStateToProps, mapDispatchToProps)(ExpenseListfilter)