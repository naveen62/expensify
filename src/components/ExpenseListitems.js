import React from 'react';
import { NavLink } from 'react-router-dom';
import moment from 'moment'
import numeral from 'numeral'

// const ExpenseListitems = ({description, amount, createdAt}) => (
//     <div>
//         <h1>{description}</h1>
//         <p>{amount} - {createdAt}</p>
//         <button>Remove</button>
//     </div>
// )
class ExpenseListitems extends React.Component {
    constructor(props) {
        super(props)
    }
    render() {
        return (
            <NavLink className='list-item' to={`/edit/${this.props.id}`} exact={true}>
                <div>
                <h1 className='list-item__title'>{this.props.description}</h1>
                <span className='list-item__sub-title'>{moment(this.props.createdAt).format('MMMM Do, YYYY')}</span>                
                </div>
                <h3 className='list-item__data'>{numeral(this.props.amount / 100).format('$0,0.00')}</h3>
            </NavLink>
        )
    }
}
export {ExpenseListitems};
export default ExpenseListitems