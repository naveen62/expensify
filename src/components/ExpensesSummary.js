import React from 'react';
import {connect} from 'react-redux';
import selectExpenses from "../selectors/expenses";
import getExpenseTotal from '../selectors/selectExpenseTotal';
import numeral from 'numeral'

class ExpensesSummary extends React.Component {

    render() {
        return (
            <div>
                <h3> Viewing {this.props.expenses.length} {this.props.expenses.length > 1 ? 'expenses' : 'expense' } totalling  {numeral(this.props.totalExpenses / 100).format('$0,0.00')} </h3>
            </div>
        )
    }
}
const mapStateToProps = (state) => {
    return {
        expenses: selectExpenses(state.expenses, state.filter),
        totalExpenses: getExpenseTotal(state.expenses)
    }
}
export default connect(mapStateToProps)(ExpensesSummary);