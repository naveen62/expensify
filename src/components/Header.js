import React from 'react';
import {Link} from  'react-router-dom';
import {connect} from 'react-redux';
import {startLogout} from '../actions/auth';

export const Header = ({startLogout}) => {
    return (
        <div>
        <header className='header' >
        <div className='content-container' >
            <div className='header__content'>
            <Link className='header__title'  to="/dashboard">
                <h1>Expensify</h1>
            </Link>
            <button className='page-header__button' onClick={startLogout}>Logout</button>
            </div>
        </div>
        </header>
        </div>
    )
}
const mapToprops = (dispatch) => {
    return {
        startLogout: () => dispatch(startLogout())
    }
}
export default connect(undefined, mapToprops)(Header);