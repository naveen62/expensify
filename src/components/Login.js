import React from 'react';
import {connect} from 'react-redux';
import {startLogin} from '../actions/auth';

const Login = (props) => {
  return(
    <div>
        <button onClick={() => {
          props.dispatch(startLogin()).then(() => {
            console.log('login succsfull')
          })
        }} >Login</button>
    </div> 
  )  
}
export default connect()(Login)

