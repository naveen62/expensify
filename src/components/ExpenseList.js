import React from 'react';
import {connect} from 'react-redux';
import ExpenseListitems from './ExpenseListitems';
import selectExpenses from '../selectors/expenses';

export const ExpenseList = (props) => (
    <div className='content-container'>
    <div className='list-header'>
        <div className='show-for-mobile'>Expenses</div>
        <div className='show-for-desc'>Expense</div>
        <div className='show-for-desc'>Amount</div>
    </div>
        <div className='list-body'>
        {props.expenses.length === 0 ? (
            <div className='list-item list-item--message   '>
             <span>No Expenses</span>
            </div>
        ) : (
            props.expenses.map((expense) => <ExpenseListitems key={expense.id} {...expense} /> )
        ) }
        </div>
    </div>
)

const mapStateToProps = (state) => {
    return {
        expenses: selectExpenses(state.expenses, state.filter)
    }
} 
export default connect(mapStateToProps)(ExpenseList);
