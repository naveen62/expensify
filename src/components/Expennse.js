import React from 'react';
import ExpenseList from './ExpenseList';
import ExpenseListfilter  from './ExpenseListfilter';
import ExpensesSummary from './ExpensesSummary';

export const Expense = () => (
    <div>
        <ExpensesSummary />
        <ExpenseListfilter />
        <ExpenseList />
    </div>
)
export default Expense;