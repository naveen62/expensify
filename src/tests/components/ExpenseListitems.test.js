import React from 'react';
import {shallow} from 'enzyme';
import {ExpenseListitems} from '../../components/ExpenseListitems';
import expenses from '../fixtures/expenses';

test('should render ExpenseListitems', () => {
    const wrapper = shallow(<ExpenseListitems {...expenses[0]} />)
    expect(wrapper).toMatchSnapshot()
})