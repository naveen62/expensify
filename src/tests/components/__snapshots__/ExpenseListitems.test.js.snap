// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpenseListitems 1`] = `
<div>
  <NavLink
    activeClassName="active"
    ariaCurrent="true"
    exact={true}
    to="/edit/1"
  >
    <h1>
      gas
    </h1>
  </NavLink>
  <p>
    $1.95
     - 
    December 31st, 1969
  </p>
</div>
`;
